lib/matplotlib/widgets.py

```diff
@@ -2006,6 +2007,42 @@ def set_visible(self, visible):
         for artist in self.artists:
             artist.set_visible(visible)
 
+    def add_artist(self, artist):
+        """Add artist to the selector."""
+        self._artists.append(artist)
+
+    @property
+    def artists(self):
+        """Tuple of the artists of the selector."""
+        return tuple(self._artists)
```

Comment:
I would like to revisit this part in a follow up PR to discuss how to use artists which get updated in a callback, need to drawn at the same time as the selector artists and play well with blitting.